src/System.Runtime.Extensions/src/System/Environment.Unix.cs

```diff
@@ -36,6 +37,7 @@ public static partial class Environment
             }
             return results;
         });
+        private static readonly bool IsMac = Interop.Sys.GetUnixName() == "OSX";
```

Comment:
Yes IsOSPlatform could (might already) return true for both Unix and OSX when running on OSX that is part of the reason it was designed the way it is.